sdk/identity/azure_identity/src/azure_pipelines_credential.rs

```diff
@@ -0,0 +1,322 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+use crate::{
+    ClientAssertion, ClientAssertionCredential, ClientAssertionCredentialOptions,
+    TokenCredentialOptions,
+};
+use azure_core::{
+    credentials::{AccessToken, Secret, TokenCredential},
+    error::ErrorKind,
+    headers::{FromHeaders, HeaderName, Headers, AUTHORIZATION},
+    HttpClient, Method, Request, StatusCode, Url,
+};
+use serde::Deserialize;
+use std::{convert::Infallible, fmt, sync::Arc};
+
+// cspell:ignore fedauthredirect oidcrequesturi
+const OIDC_VARIABLE_NAME: &str = "SYSTEM_OIDCREQUESTURI";
+const OIDC_VERSION: &str = "7.1";
+const TFS_FEDAUTHREDIRECT_HEADER: HeaderName = HeaderName::from_static("x-tfs-fedauthredirect");
+
+// TODO: https://github.com/Azure/azure-sdk-for-rust/issues/682
+const ALLOWED_HEADERS: &[&str] = &["x-msedge-ref", "x-vss-e2eid"];
+
+#[derive(Debug)]
+pub struct AzurePipelinesCredential(ClientAssertionCredential<Client>);
+
+/// Options for constructing a new [`AzurePipelinesCredential`].
+#[derive(Debug, Default)]
+pub struct AzurePipelinesCredentialOptions {
+    /// Options for the [`ClientAssertionCredential`] used by the [`AzurePipelinesCredential`].
+    pub credential_options: ClientAssertionCredentialOptions,
```

Comment:
@demoray see here and related code for an example of what I was referring to in #2309